x-pack/plugins/infra/server/lib/alerting/inventory_metric_threshold/inventory_metric_threshold_executor.ts

```diff
@@ -245,6 +254,8 @@ export const createInventoryMetricThresholdExecutor = (libs: InfraBackendLibs) =
       const recoveredAlertId = alert.getId();
       const indexedStartedDate = getAlertStartedDate(recoveredAlertId) ?? startedAt.toISOString();
       const alertUuid = getAlertUuid(recoveredAlertId);
+      const alertHits = alertUuid ? await getAlertByAlertUuid(alertUuid) : undefined;
```

Comment:
`ruleDataClient.getReader().search()` will log and throw any errors during querying which will bubble back up to here and cause the rule execution to end in an error. I would suggest either adding a try/catch here around `getAlertByAlertUuid` or a try/catch within `getAlertByAlertUuid` to gracefully handle potential ES errors.

Reply:
good catch! 👍 I have added try/catch in `getAlertByAlertUuid()`. I haven't logged anything as the error would be already logged by 'search()' function.